src/hooks/useRender/index.ts

```diff
@@ -0,0 +1,100 @@
+import { useCallback, useEffect, useState } from 'react';
+
+interface Wave {
+  color: string,
+  width: number,
+}
+
+interface Text {
+  font: string;
+  align: CanvasTextAlign,
+  value: string,
+}
+
+interface Prameters {
+  autoinit?: boolean;
+  bufferLength: number;
+  background?: string;
+  wave?: Wave;
+  text?: Text
+}
+
+function useRender(config: Prameters) {
+  const [canvas, setCanvas] = useState<HTMLCanvasElement>();
+  const [context, setContext] = useState<CanvasRenderingContext2D>();
+  const [bufferLength, setBufferLength] = useState<number>(config.bufferLength || 0);
+  const [buffer, setBuffer] = useState<Uint8Array>(new Uint8Array(bufferLength));
+  const [text, setText] = useState<string>(config.text ? config.text.value : 'empty');
+  const [update, setUpdate] = useState<() => void>(() => {});
+
+  const draw = useCallback(() => {
+    if (context) {
+      requestAnimationFrame(draw);
+      update();
+      const textWidth = context.measureText(text).width + 30;
+      const textHeight = 50;
+      const textBackground = {
+        x: (canvas.width / 2) - (textWidth / 2),
+        y: (canvas.height / 2) - (textHeight / 2),
+      };
+
+      context.fillStyle = config.background || 'rgba(255, 255, 255)';
```

Comment:
Acho que cada que for desenha no canvas deveria ter um método próprio. Assim teriamos uma sequência de algo como: "drawBackground()", "drawLine()", ...